meilisearch-lib/src/index/search.rs

```diff
@@ -97,8 +102,13 @@ impl Index {
             search.query(query);
         }
 
-        search.limit(query.limit);
-        search.offset(query.offset.unwrap_or_default());
+        // Make sure that a user can't get more documents than the hard limit,
+        // we align that on the offset too.
+        let offset = min(query.offset.unwrap_or(0), HARD_RESULT_LIMIT);
+        let limit = min(query.limit, HARD_RESULT_LIMIT.saturating_sub(offset));
```

Comment:
```suggestion
        let limit = query.limit.min(HARD_RESULT_LIMIT.saturating_sub(offset));
```

the saturating sub is not strictly necessary since we already clamped `offset` to `HARD_RESULT_LIMIT`, but I don't see any downsides, so cool

Reply:
Yeah you are right about the `saturating_sub` uselessness and I'll keep it :)